.github/actions/scenarios/backend/external-agent/action.yml

```diff
@@ -0,0 +1,235 @@
+        -Dgateway.nettyPort=6894 \
+        -Dheartbeat.interval=5000 \
+        -jar sermant-agent-${{ env.sermantVersion }}/agent/agentcore-test-application-1.0.0-jar-with-dependencies.jar > ${{ env.logDir }}/agentcore-test.log 2>&1 &
+    - name: waiting for application start
+      shell: bash
+      run: |
+        bash ./sermant-integration-tests/scripts/checkService.sh http://127.0.0.1:8915/ping 120
+    - name: test start with sermant agent without external agent
+      shell: bash
+      run: |
+        export TEST_MODE=startWithoutExternalAgent
+        python -m unittest ./sermant-integration-tests/scripts/test_backend_external_agent.py
+    - name: stop application
+      shell: bash
+      run: |
+        netstat -nlp | grep :8915 | awk '{print $7}' | awk -F "/" '{print $1}' | xargs kill
+        sleep 3
+    - name: start application with sermant agent and external agent
+      shell: bash
+      env:
+        dynamic.config.serverAddress: 127.0.0.1:2181
+        dynamic.config.dynamicConfigType: ZOOKEEPER
+        service.meta.project: TestAgentCore
+      run: |
+        nohup java -javaagent:sermant-agent-${{ env.sermantVersion }}/agent/sermant-agent.jar \
+        -Dserver.port=8916 \
+        -Dagent.config.externalAgent.injection=true \
+        -Dagent.config.externalAgent.name=OTEL \
+        -Dagent.config.externalAgent.file=opentelemetry-javaagent.jar \
+        -Dagent.service.heartbeat.enable=true \
+        -Dagent.service.gateway.enable=true \
+        -Devent.enable=true \
+        -Dgateway.nettyPort=6894 \
+        -Dheartbeat.interval=5000 \
+        -jar sermant-agent-${{ env.sermantVersion }}/agent/agentcore-test-application-1.0.0-jar-with-dependencies.jar > ${{ env.logDir }}/agentcore-test.log 2>&1 &
+    - name: waiting for application start
+      shell: bash
+      run: |
+        bash ./sermant-integration-tests/scripts/checkService.sh http://127.0.0.1:8916/ping 120
+    - name: test start with sermant agent and external agent
+      shell: bash
+      run: |
+        export TEST_MODE=startWithExternalAgent
+        python -m unittest ./sermant-integration-tests/scripts/test_backend_external_agent.py
+    - name: stop application
+      shell: bash
+      run: |
+        netstat -nlp | grep :8916 | awk '{print $7}' | awk -F "/" '{print $1}' | xargs kill
+        sleep 3
+    - name: start application alone
+      shell: bash
+      env:
+        dynamic.config.serverAddress: 127.0.0.1:2181
+        dynamic.config.dynamicConfigType: ZOOKEEPER
+        service.meta.project: TestAgentCore
+      run: |
+        nohup java -Dserver.port=8917 \
+        -jar sermant-agent-${{ env.sermantVersion }}/agent/agentcore-test-application-1.0.0-jar-with-dependencies.jar > ${{ env.logDir }}/agentcore-test.log 2>&1 &
+    - name: waiting for application start
+      shell: bash
+      run: |
+        bash ./sermant-integration-tests/scripts/checkService.sh http://127.0.0.1:8917/ping 120
+        sleep 10
```

Comment:
Why sleep 10 seconds here?

Reply:
make sure the application have sent heartbeat meaasge to Backend